### ISSUE TITLE
[iOs] In-App purchase works in TestFlight, but not in the productive app.

### ISSUE THREAD

**Issue by Schwankenson** [j3k0/cordova-plugin-purchase#1273]
### system info

iOs 15.1 in iPad (2018)

### Expected behavior

store.order() should start the purchase and open the app store purchase pop up 

### Observed behavior

It works perfectly in TestFlight and when I compile and install on the device with XCode. But now the app is productive and in "Ready for sales mode" in app store. And Nothing happens, if user tries to in app purchase (auto-renewing and non-consumables)

Any idea how to debug this? Any idea why this could happen? It works, when I install the same version of my app in TestFlight on the same device.

**Comment by zeromark223** [comment]
**system info :** 
IOS 15.2 Iphone 8.

I meet the same problem.
It works perfectly in TestFlight with TestFlight Account.
When I submit to apple review, the product can't register then product can't use by store.order().

**Comment by Schwankenson** [comment]
My app is even in state "Ready for sale". The Appstore review team tested in sandbox mode, too. But The purchase does not start on the app, which is live on the Appstore.

**Comment by Schwankenson** [comment]
Ok, fixed it:
I have a custom coded receipt validation. In production mode, in-app-purchase-2 validates receipt on load (before any purchase was done). This receipt content does not contain the key `latest_receipt_info`. But I accessed it in my server without checking if it exists. This caused the error.

So test your receipt validation, if you have the same problem like I have.

I used https://www.charlesproxy.com/ to track the traffic from my production app.
